server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/JMAPAPIRoute.scala

```diff
@@ -0,0 +1,30 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+package org.apache.james.jmap.routes
+
+import org.apache.james.jmap.model.Invocation
+import org.apache.james.jmap.model.Invocation.MethodName
+import org.reactivestreams.Publisher
+
+trait JMAPAPIRoute[T] {
+  val methodName: MethodName
+
+  def process(invocation: Invocation): Publisher[T]
```

Comment:
From the spec I don't think we can return anything but an invocation. Can we then replace `Publisher[T]` by `Publisher[Invocation]` ?

Reply:
That correct.